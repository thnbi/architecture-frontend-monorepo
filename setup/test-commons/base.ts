import type {Config} from '@jest/types';
import path from 'path';

// Sync object
const config: Config.InitialOptions = {
    testEnvironment: 'jsdom',
    verbose: true,
    transform: {
        '\\.[jt]sx?$': ['babael-jest', {
            configFile: path.resolve(_dirname, '.babelrc'),
        }],
    }
};

export default config;
