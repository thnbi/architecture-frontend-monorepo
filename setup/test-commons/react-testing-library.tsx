import { render as rtlRender } from '@testing-library/react';

function DefaultWrapper({ children }) {
  return children;
}

export const customRender = (AllTheProviders = DefaultWrapper) => {
  return (ui) => {
    return rtlRender(ui, { wrapper: AllTheProviders });
  }
}

export * from '@testing-library/react';

export const render = () => console.error('Please create your own render method using customRender')
