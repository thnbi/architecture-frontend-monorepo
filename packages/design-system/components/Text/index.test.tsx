import { Text } from "./index";
import { customRender } from "@thnbi/test-commons/react-testing-library";

const render = customRender();

describe("<Text />", () => {
  it("renders h1 tag", () => {
    const { container } = render(<Text tag="h1">Hello World</Text>);
    expect(container).toMatchSnapshot();
  });
});
