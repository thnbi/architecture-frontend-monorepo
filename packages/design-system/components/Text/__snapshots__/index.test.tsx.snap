// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders h1 tag 1`] = `
<div>
  <h1
    class="sc-bcXHqe cQPymi"
  >
    Hello World
  </h1>
</div>
`;
